client/tsconfig.json

```diff
@@ -16,13 +16,6 @@
 			"calypso/*": [ "./*" ]
 		}
 	},
-	"references": [
-		{ "path": "../packages/calypso-analytics" },
-		{ "path": "../packages/components" },
-		{ "path": "../packages/data-stores" },
-		{ "path": "../packages/language-picker" },
-		{ "path": "../packages/react-i18n" }
-	],
```

Comment:
There's also a `packages/languages` package that is written in TypeScript and isn't being migrated in this PR: it's still built during the `prepare` step.

Reply:
`packages/languages` is migrated in the sense that has `calypso:src`, but still requires a build step because we do use it to build Calypso: `yarn build-languages` calls `bin/build-languages.js`  which requires `'@automattic/languages'`

Reply:
Oh, I see, the package is used by a Node.js build script and needs to be built 🙁

Reply:
The `@automattic/languages` is in a very similar situation as a Gutenberg webpack plugin we're discussing with @sirreal in https://github.com/WordPress/gutenberg/pull/26382#discussion_r525964214

I think we'd be better off if the package was written in plain JavaScript, where it would be a trivial two-liner:
```js
import data from './languages-meta.json';
export default Object.values( data );
```
and if the types were in a separate `index.d.ts` file.

Then the package wouldn't need any build step at all, could be directly consumed by Node.js build scripts, and consumers would still get TypeScript types for it without any observable change.

This approach is an exception rather that a rule, but I think it would be justified for certain packages like these I mention.